velox/functions/lib/Slice.cpp

```diff
@@ -77,12 +85,15 @@ class SliceFunction : public exec::VectorFunction {
       BaseVector::flattenVector(args[0]);
     }
 
-    VectorPtr localResult =
-        applyArray<int64_t>(rows, args, context, outputType);
+    VectorPtr localResult = kind_ == TypeKind::INTEGER
+        ? applyArray<int32_t>(rows, args, context, outputType)
+        : applyArray<int64_t>(rows, args, context, outputType);
     context.moveOrCopyResult(localResult, rows, result);
   }
 
  private:
+  /// kind The type kind of start and length.
```

Comment:
nit: use `//` for single-line comment, and remove the `kind` before the sentence.